tests/test_driver.py

```diff
@@ -75,7 +75,7 @@ def test_ensure(self):
         self.assertFalse(relay1.inverted)
 
         # Test retrieving the existing relay with the same pin and inversion
-        relay2 = Driver.ensure(17, True, MockFactory())
+        relay2 = Driver.ensure(17, False, MockFactory())
```

Comment:
@cesaregiannetti , the purpose of this line is to ensure that even with inversion (`True`) the `ensure()` method does not create a new instance, but reuses the existing one.
It's required for the cases when user changes the corresponding setting.
Please revert the change of this line

```suggestion
        relay2 = Driver.ensure(17, True, MockFactory())
```

Reply:
I think you're talking about line 83, where relay3 is inverted, whereas here the comment says
> # Test retrieving the existing relay with the same pin and inversion
so I think relay2 is actually testing just the cache

Reply:
I also adjusted the other tests to reflect the change. there's no need to have pin state and relay state anymore, as they are the same, thanks to gpiozero interface, but I kept them for clarity

Reply:
ok